test/org/sagebionetworks/bridge/models/accounts/SignInTest.java

```diff
@@ -89,5 +89,23 @@ public void acceptsUsernameAsEmail() throws Exception {
         assertEquals("studyValue", signIn.getStudyId());
         assertEquals("tokenValue", signIn.getToken());
         assertEquals("reauthTokenValue", signIn.getReauthToken());
-    }  
+    }
+    
+    @Test
+    public void signInAccountIdWithEmail() {
+        SignIn signIn = new SignIn.Builder().withStudy(TestConstants.TEST_STUDY_IDENTIFIER).withEmail("email")
+                .withPassword("password").build();
+        AccountId accountId = signIn.getAccountId();
+        assertEquals(TestConstants.TEST_STUDY_IDENTIFIER, accountId.getStudyId());
+        assertEquals("email", accountId.getEmail());
+    }
+    
+    @Test
+    public void signInAccountIdWithPhone() {
+        SignIn signIn = new SignIn.Builder().withStudy(TestConstants.TEST_STUDY_IDENTIFIER)
+                .withPhone(TestConstants.PHONE).withPassword("password").build();
+        AccountId accountId = signIn.getAccountId();
+        assertEquals(TestConstants.TEST_STUDY_IDENTIFIER, accountId.getStudyId());
+        assertEquals(TestConstants.PHONE.getNumber(), accountId.getPhone().getNumber());
+    }
```

Comment:
Is it worth adding a case for getAccountId() with neither email nor phone?

Reply:
Added. Verified we validate before calling getAccountId() in all pathways since this will generate a server error.